library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java

```diff
@@ -378,30 +378,50 @@
     private final ContentResolver contentResolver;
     private final DownloadsUriProvider downloadsUriProvider;
     private final SystemFacade systemFacade;
+    private final BatchRepository batchRepository;
 
     private Uri baseUri;
 
-    public DownloadManager(Context context, ContentResolver resolver) {
-        this(context, resolver, DownloadsUriProvider.getInstance(), new RealSystemFacade(context), false);
+    public static DownloadManager newInstance(Context context, ContentResolver contentResolver) {
```

Comment:
It would be something like

```java
public DownloadManager(Context context, ContentResolver resolver) {
        this(context,
             resolver,
             DownloadsUriProvider.getInstance(),
             new RealSystemFacade(context),
             new BatchRepository(resolver, 
                                 new DownloadDeleter(resolver), 
                                 DownloadsUriProvider.getInstance(), 
                                 new RealSystemFacade(context)),
             false);
    }

    public DownloadManager(Context context, ContentResolver contentResolver, boolean verboseLogging) {
        this(context,
             contentResolver,
             DownloadsUriProvider.getInstance(),
             new RealSystemFacade(context),
             new BatchRepository(contentResolver, 
                                 new DownloadDeleter(contentResolver), 
                                 DownloadsUriProvider.getInstance(), 
                                 new RealSystemFacade(context)),
             verboseLogging);
    }

    DownloadManager(Context context, ContentResolver resolver, DownloadsUriProvider downloadsUriProvider) {
        this(context,
             resolver,
             downloadsUriProvider,
             new RealSystemFacade(context),
             new BatchRepository(resolver, 
                                 new DownloadDeleter(resolver), 
                                 downloadsUriProvider, 
                                 new RealSystemFacade(context)),
             false);
    }

    DownloadManager(Context context,
                    ContentResolver contentResolver,
                    DownloadsUriProvider downloadsUriProvider,
                    SystemFacade systemFacade,
                    BatchRepository batchRepository,
                    boolean verboseLogging) {
        this.contentResolver = contentResolver;
        this.downloadsUriProvider = downloadsUriProvider;
        this.baseUri = downloadsUriProvider.getContentUri();
        this.systemFacade = systemFacade;
        this.batchRepository = batchRepository;
        GlobalState.setContext(context);
        GlobalState.setVerboseLogging(verboseLogging);
    }
```

Notice the double `new RealSystemFacade)` and `DownloadsUriProvider.getInstance()`